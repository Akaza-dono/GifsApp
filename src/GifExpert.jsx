import React, { useState } from "react";
import {AddCategory, GifGrid} from './components'

export const GifExpert = () => {
  const [categorias, setCategorias] = useState(["One Punch"]);

  const onAddnewCategory = (value) => {
    if (!categorias.includes(value)) {
      setCategorias([value, ...categorias]);
    }
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onNewValue={onAddnewCategory} />

      {categorias.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
