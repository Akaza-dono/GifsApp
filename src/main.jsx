import React from 'react';
import ReactDOM from 'react-dom/client';
import {GifExpert} from './GifExpert';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpert/>
  </React.StrictMode>
);
