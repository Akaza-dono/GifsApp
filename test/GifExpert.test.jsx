import { render, screen } from "@testing-library/react"
import { GifExpert } from "../src/GifExpert"

describe("Testsuit de gifExpert",()=>{
    test("Debe hacer match con el snapshot",()=>{
        const {container} = render(<GifExpert/>);
        screen.debug()
        expect(container).toMatchSnapshot();

    });
    // test("debe de existir el input con el value indefinido",()=>{
    //     expect(screen.getByText('Cargando...')).toBeTruthy()
    //     // const input = screen.getByRole('textbox');
    //     // screen.debug()    
    // })
})