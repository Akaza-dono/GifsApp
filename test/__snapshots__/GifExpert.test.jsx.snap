// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testsuit de gifExpert Debe hacer match con el snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
